pkg/postgres/postgres.go

```diff
@@ -51,7 +59,10 @@ func Wait(service, dsn string) *DB {
 	stream.Close()
 	// TODO(titanous): handle discoverd disconnection
 
-	db, err := Open(service, dsn)
+	// TODO(jpg): Race condition between DB registering with discoverd and being able
+	// to authenticate with password. Resolve this correctly before merging.
```

Comment:
That would require pulling the pgmanager client into pkg/postgres though. Which probably means we should use it to wait for read/write state as well. Not sure if it's a good idea to do all of that in this change set though, it's already getting unwieldy and hard to track down bugs.
This along with prepared statements and changing fixing up the deployment transactions etc I think should go in after.